wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/linking/expression/files/EngineFunctionEvaluators.scala

```diff
@@ -0,0 +1,216 @@
+package wdl.draft3.transforms.linking.expression.files
+
+import cats.syntax.apply._
+import cats.syntax.traverse._
+import cats.syntax.validated._
+import cats.instances.list._
+import common.validation.ErrorOr._
+import common.validation.ErrorOr.ErrorOr
+import common.validation.Validation._
+import wdl.draft3.transforms.linking.expression.values.EngineFunctionEvaluators.processValidatedSingleValue
+import wdl.model.draft3.elements.ExpressionElement
+import wom.format.MemorySize
+import wdl.draft3.transforms.linking.expression.values._
+import wdl.model.draft3.elements.ExpressionElement._
+import wdl.model.draft3.graph.expression.{FileEvaluator, ValueEvaluator}
+import wdl.model.draft3.graph.expression.ValueEvaluator.ops._
+import wdl.model.draft3.graph.expression.FileEvaluator.ops._
+import wdl.shared.transforms.evaluation.values.EngineFunctions
+import wdl4s.parser.MemoryUnit
+import wom.expression.IoFunctionSet
+import wom.types._
+import wom.values.WomArray.WomArrayLike
+import wom.values.{WomArray, WomFile, WomFloat, WomInteger, WomOptionalValue, WomPair, WomPrimitive, WomSingleFile, WomString, WomValue}
+import wom.types.coercion.ops._
+import wom.types.coercion.defaults._
+import wom.types.coercion.WomTypeCoercer
+
+import scala.concurrent.duration._
+import scala.concurrent.{Await, Future}
+import scala.util.{Success, Try}
+
+object EngineFunctionEvaluators {
+
+  private def evaluateToFile(forFunction: String, a: ExpressionElement, inputs: Map[String, WomValue], ioFunctionSet: IoFunctionSet): ErrorOr[WomSingleFile] = {
+    a.evaluateValue(inputs, ioFunctionSet) flatMap {
+      case p: WomPrimitive => WomSingleFile(p.valueString).validNel
+      case other => s"Could not predict files to delocalize from '$a' for $forFunction. Expected a primitive but got ${other.getClass.getSimpleName}".invalidNel
+    }
+  }
+
+  implicit val stdoutFunctionEvaluator: FileEvaluator[StdoutElement.type] = new FileEvaluator[StdoutElement.type] {
```

Comment:
[ToL] I feel like there must be a clever way to reduce duplication here - type parameterization perhaps?